Sources/CreateAPI/Generator/Templates.swift

```diff
@@ -343,10 +346,23 @@ final class Templates {
         """
     }    
     
-    func encodeOneOf(properties: [Property]) -> String {
-        let statements = properties.map {
-            "case .\($0.name)(let value): try container.encode(value)"
+    func encodeOneOf(properties: [Property], discriminator: Discriminator?) -> String {
+        var statements: [String] = []
+        if let discriminator = discriminator {
+            for property in properties {
+                let correspondingMappings = discriminator.mapping.filter { $1 == property.type }.sorted { $0.key < $1.key }
```

Comment:
I see that this has been repeated three times so do you think we can extract it out to a computed property somewhere? Maybe on `Discriminator`?

Reply:
Thanks for your reviews! I couldn't find a way to make it with a computed property so I went with a function. Is this alright?
[446ea60](https://github.com/CreateAPI/CreateAPI/pull/36/commits/446ea605b9304ba50c7397a8ad97239f38ea44c8)